doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,382 @@
+And the FABRID option that contains the FABRID hopfield metadata fields and a path validator field.
+The Identifier option can be used without the FABRID option and can therefore also be used by other extensions.
+The FABRID option on the other hand requires that the Identifier option is specified in the HBH extension before the FABRID option.
+
+.. _identifier-option:
+
+Identifier Option
+^^^^^^^^^^^^^^^^^^
+
+The Identifier Option always has a length of 8 bytes and look like::
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+                                    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                                    |  OptType = 3  |  OptLen = 8   |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |R R R R R|                Timestamp                            |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                          Packet ID                            |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Timestamp
+    The 27 bit timestamp of when the packet has been sent with 1 millisecond precision
+    relative to the timestamp of the first InfoField of the SCION header.
+Packet ID
+    The 32 bit packet ID that is used together with the timestamp to uniquely identify
+    the packet originating from a particular flow.
+
+.. _fabrid-option:
+
+FABRID Option
+^^^^^^^^^^^^^^
+
+The FABRID Option has a length of (#NumberOfOnPathASes + 1)*4 bytes.
+This hop-by-hop option has an allignment of 4 bytes::
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+                                    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                                    |  OptType = 4  |  OptLen = ?   |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |    ....       | | |               ....                        |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                       Path Validator                          |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Encrypted PolicyID
+    The 8 bit encrypted FABRID policy index.
+F
+    Stands for “FABRID enabled” and if this is set to false, the router responsible for
+    that hop will not apply any FABRID logic to this packet.
+    This can be used e.g. if an on-path AS does not support FABRID.
```

Comment:
Would be nice to think about a more extensive list when this will be used.
- when an on-path AS supports FABRID, but the end host does not care
- others?